assets/src/edit-story/app/story/useStoryReducer/reducers/updateElements.js

```diff
@@ -70,16 +70,51 @@ function updateElements(
     return state;
   }
 
-  const updatedElements = oldPage.elements.map((element) => {
-    if (!idsToUpdate.includes(element.id)) {
-      return element;
+  const {
+    animationLookup,
+    elements: updatedElements,
+  } = oldPage.elements.reduce(
+    ({ animationLookup, elements }, element) => {
+      if (!idsToUpdate.includes(element.id)) {
+        return {
+          animationLookup,
+          elements: [...elements, element],
+        };
+      }
+
+      const { animation, ...elem } = updateElementWithUpdater(
+        element,
+        propertiesOrUpdater,
+        pageIndex
+      );
+
+      const animLookup = animation
+        ? { [animation.id]: { ...animation, targets: [elem.id] } }
+        : {};
+
+      return {
+        animationLookup: {
+          ...animationLookup,
+          ...animLookup,
+        },
+        elements: [...elements, elem],
+      };
+    },
+    {
+      animationLookup: {},
+      elements: [],
     }
-    return updateElementWithUpdater(element, propertiesOrUpdater, pageIndex);
-  });
+  );
+
+  const newAnimations =
+    Object.keys(animationLookup).length > 0
+      ? updateAnimations(oldPage.animations || [], animationLookup)
+      : oldPage.animations;
 
   const newPage = {
     ...oldPage,
     elements: updatedElements,
+    ...(newAnimations ? { animations: newAnimations } : {}),
```

Comment:
Only add `animation` prop if we have actual animation data (keeps us from having to update a bunch of tests with `animation: undefined`).